compiler/stack_build.cmake

```diff
@@ -11,7 +11,7 @@
 cmake_policy (SET CMP0012 NEW)
 
 execute_process (
-    COMMAND ${STACK_EXECUTABLE} setup
+    COMMAND ${STACK_EXECUTABLE} setup ${BOND_STACK_OPTIONS}
```

Comment:
`stack` has two different kinds of options: those that go between `stack` and the command (like `build` or `setup`) and those that go with a corresponding command (like `--allow-different-user` that the `setup` command [among others?] takes). This appears to only be for command-specific options, but is used across many different commands.

Which stack commands need `--allow-different-user`?

Reply:
I think `--allow-different-user` is not specific to `setup`. I can see it in `stack --help`.

Reply:
Yup. I misread the help. This should then be:

```
COMMAND ${STACK_EXECUTABLE} ${BOND_STACK_OPTIONS} setup
```

Reply:
Updated.